conan/tools/cmake/cmake_initial_cache.py

```diff
@@ -0,0 +1,74 @@
+import os
+
+from conan.tools.cmake.presets import load_cmake_presets, get_configure_preset
+from conans.util.files import save, save_append
+
+
+def _format_cache_variables(cache_variables):
+    ret = []
+    for name, value in cache_variables.items():
+        if isinstance(value, bool):
+            type_ = "BOOL"
+            v = "ON" if value else "OFF"
+        elif value in ("ON", "OFF"):
+            type_ = "BOOL"
+            v = value
+        else:
+            type_ = "STRING"
+            v = f'"{value}"'
+        ret.append(f'set({name} {v} CACHE {type_} "" FORCE)')
+    return "\n".join(ret)
+
+
+def get_initial_cache_path(conanfile):
+    return os.path.join(conanfile.generators_folder, "CMakeInitialCache.cmake")
```

Comment:
Use a ``conan_initial_cache.cmake``, more aligned with the ``conan_toolchain.cmake`` naming